Doesn't compile with the Linux Static SDK (aka musl)

### Expected behavior

Now that the [Static Linux SDK](https://www.swift.org/documentation/articles/static-linux-getting-started.html) has been released, it open the doors to cross-paltform compilation from Mac that doesn't require Docker or depend on Amazon Linux.  
In theory the binary created with the Static Linux SDK should work "as is" on AWS Lambda.

To make this happen, the AWS Lambda Runtime must slightly adjust the way it detects Linux and import Glibc.

I'll send a PR to fix this in a moment.

### Actual behavior

Compilation errors such as 

```
swift-aws-lambda-runtime/Sources/AWSLambdaRuntimeCore/Lambda.swift:16:8: error: no such module 'Glibc'
 14 | 
 15 | #if os(Linux)
 16 | import Glibc
    |        `- error: no such module 'Glibc'
 17 | #else
 18 | import Darwin.C
``` 

### Steps to reproduce

1. Install the Linux Static SDK ([doc](https://www.swift.org/documentation/articles/static-linux-getting-started.html))
2. Apply [the patch as described in this message](https://forums.swift.org/t/distributing-binaries-for-linuxes-static-linking/21366/24)
3. Compile with `swift build --swift-sdk aarch64-swift-linux-musl`

### If possible, minimal yet complete reproducer code (or URL to code)

_No response_

### What version of this project (`swift-aws-lambda-runtime`) are you using?

1.0.0-alpha1

### Swift version

```
✗ swift --version && uname -a 
Apple Swift version 6.0-dev (LLVM 6e13b8be259f20f, Swift 372de4670b9cd55)
Target: arm64-apple-macosx14.0
Darwin bcd07458a91c 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000 arm64
```

### Amazon Linux 2 docker image version

_No response_

--- a/Package.swift
+++ b/Package.swift
@@ -21,8 +21,8 @@ let package = Package(
         .library(name: "AWSLambdaTesting", targets: ["AWSLambdaTesting"]),
     ],
     dependencies: [
-        .package(url: "https://github.com/apple/swift-nio.git", .upToNextMajor(from: "2.43.1")),
-        .package(url: "https://github.com/apple/swift-log.git", .upToNextMajor(from: "1.4.2")),
+        .package(url: "https://github.com/apple/swift-nio.git", .upToNextMajor(from: "2.67.0")),
+        .package(url: "https://github.com/apple/swift-log.git", .upToNextMajor(from: "1.5.4")),
         .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
     ],
     targets: [

--- a/Plugins/AWSLambdaPackager/Plugin.swift
+++ b/Plugins/AWSLambdaPackager/Plugin.swift
@@ -16,8 +16,16 @@ import Dispatch
 import Foundation
 import PackagePlugin
 
-#if canImport(Glibc)
+#if os(macOS)
+import Darwin
+#elseif canImport(Glibc)
 import Glibc
+#elseif canImport(Musl)
+import Musl
+#elseif os(Windows)
+import ucrt
+#else
+#error("Unsupported platform")
 #endif
 
 @main

--- a/Sources/AWSLambdaRuntimeCore/Lambda.swift
+++ b/Sources/AWSLambdaRuntimeCore/Lambda.swift
@@ -12,10 +12,16 @@
 //
 //===----------------------------------------------------------------------===//
 
-#if os(Linux)
-import Glibc
-#else
+#if os(macOS)
 import Darwin.C
+#elseif canImport(Glibc)
+import Glibc
+#elseif canImport(Musl)
+import Musl
+#elseif os(Windows)
+import ucrt
+#else
+#error("Unsupported platform")
 #endif
 
 #if swift(<5.9)

